File: C/Code/1500/1546 - 평균.c
#include <stdio.h>

int main(n, temp) {
	float max = -1, sum = 0;
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		scanf("%d", &temp);								// i'th subject score
		sum += temp, max = (max > temp) ? max : temp;	// if score > max, max will change
	}
	printf("%f", sum / max * 100 / n);					// print new average
	return 0;
}

File: Java/Code/1500/1546 - 평균.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int subject = sc.nextInt(), max = -1;
		double average = 0;
		for (int i = 0; i < subject; i++) {
			int temp = sc.nextInt();
			if (temp > max)
				max = temp;
			average += temp;
		}
		average = average / subject / max * 100;
		System.out.println(String.format("%.3f", average));
		return;
	}
}